robottelo/report_portal/portal.py

```diff
@@ -0,0 +1,221 @@
+        )
+        typed_launches = list(
+            filter(lambda launch: launch['name'].lower() == launch_type, all_launches)
+        )
+        for launch_info in typed_launches[:latest]:
+            version_compiler = re.compile(r'([\d\.]+)[\.-](\d+\.\d|[A-Z]+)')
+            launch_name = next(filter(version_compiler.search, launch_info['tags']))
+            sat_ver = re.search(version_compiler, launch_name).group(1)
+            snap_version = re.search(version_compiler, launch_name).group(2)
+            if sat_ver in launches_:
+                launches_[sat_ver].update({snap_version: launch_info})
+            else:
+                launches_[sat_ver] = {snap_version: launch_info}
+        if sat_version:
+            if sat_version not in launches_:
+                raise ValueError(
+                    f'The given satellite version \'{sat_version}\' is not available in '
+                    f'Report portal project \'{self.rp_project}\''
+                )
+            else:
+                return launches_[sat_version]
+        return launches_
+
+    def launch(self, sat_version, snap_version=None, launch_type='satellite6'):
+        """Returns a specific launch data in Report Portal Project
+
+        This does not includes each tests data in launch
+
+        :param str sat_version: The satellite version
+        :param str snap_version: The snap version of a given satellite version
+            if None, the latest launch data of a given sat_version is returned
+        :param str launch_type: Either satellite6 or upgrades, default returns only
+            non-upgrade launches
+        :returns dict: The data directory of requested or latest launch
+        """
+        launches_ = self.launches(sat_version=sat_version, launch_type=launch_type)
+        if snap_version:
+            if snap_version not in launches_.keys():
+                raise ValueError(
+                    f'The given snap_version \'{snap_version}\' is not available '
+                    f'in satellite version \'{sat_version}\''
+                )
+        else:
+            # Get the latest launch of the given sat_version
+            snap_version = list(launches_.keys())[0]
+        return launches_[snap_version]
+
+
+class Launch:
+    def __init__(self, rp, sat_version, snap_version=None, launch_type='satellite6'):
+        """Initiates the Launch and its properties of a specified snap version in
+        a specified satellite version
+
+        :param ReportPortal rp: An object of a Report Portal Class
+        :param str sat_version: The satellite version
+        :param str snap_version: The snap version of a given satellite version
+            if None, the latest launch data of a given sat_version is considered
+        :param str launch_type: Either satellite6 or upgrades, default returns only
+            non-upgrade launches
+        """
+        self.report_portal = rp
+        self.defect_types = {
```

Comment:
This appears static, it can be a class attribute, not an instance attribute.  I think its also an attribute/property of ReportPortal, not of a launch. Its the same for all launches on a ReportPortal instance.

Reply:
Its also being redefined in #7758 in constants. Its relative to ReportPortal, I think it should be a class attribute there.

Reply:
Because it's more specific to launch tests I thought to keep it here. But still, I disagree with making it ReportPortal attributes. Let's say we cannot filter launches of Report portal with defect types but tests of launches can be.

The reason I wanted to keep it in constants as I am using them in rerun plugin in #7758 and that is even before RP class initialization.

If we don't want to define it as constants then its better to keep them as RP class attributes so that we can quickly fail in options sanitization in #7758 and that is even before intiating launches objects.

So doing it !